src/components/Recovery.jsx

```diff
@@ -141,12 +162,20 @@ function Recovery({ credentials, setCredentials }) {
         <input
           name="password"
           type="password"
-          defaultValue={credentials ? credentials.password : ""}
+          value={password}
+          onChange={handlePasswordChange}
         />
         <br />
+        <span style={{ color: "red" }}>{passwordError}</span>
+        <br />
```

Comment:
It works the same since we are using passwordError here which is the output of validatePassword function (import { validatePassword } from "../constants/passwordPolicy";).

Reply:
I meant that text should be added *for the user to see* what constitutes a valid password.